### PR TITLE
[FE-215] Levels section

### PR DESCRIPTION
**Pull request by AnCichocka**
https://oops2000.atlassian.net/browse/SCRUM-215

Pretty big pr. Some notes:
- styling is not done - pretend that width changes in forms do not exist, I am aware of that
- grade check needs to be implemented in `follow up PR`, please do not update this one
- validation should works, but it needs some polish

### CODE REVIEW

File: frontend/src/components/Edition/Sections/LevelsSection/AddSetForm/AddLevelForm.tsx [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+
+      // validate with already added levels
+      const { nameError, maxPointsError, gradeError } =
+        validateWithAddedLevels(values);
+
+      errors = nameError ? { ...errors, name: nameError } : errors;
+      errors = maxPointsError
+        ? { ...errors, maxPoints: maxPointsError }
+        : errors;
+      errors = gradeError ? { ...errors, grade: gradeError } : errors;
```
Comment (AnCichocka) [inline frontend/src/components/Edition/Sections/LevelsSection/AddSetForm/AddLevelForm.tsx:71]: place prepared for grade validation - it needs to be implemented in validateWithAddLevels function and errors should display just fine

File: frontend/src/components/Edition/Sections/LevelsSection/AddSetForm/AddSetForm.tsx [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+    // check if max points not duplicated
+    const maxPointsError = levels.find((l) => l.maxPoints === values.maxPoints)
+      ? "duplicated maxPoints"
+      : undefined;
+
+    // TODO: check if grade is consistent
+    let gradeError: string | undefined;
+
+    return {
```
Comment (AnCichocka) [inline frontend/src/components/Edition/Sections/LevelsSection/AddSetForm/AddSetForm.tsx:117]: implement grade check here

File: frontend/src/components/Edition/Sections/LevelsSection/AddSetForm/AddSetForm.tsx [hunk 3]
```diff
@@ -111,13 +112,33 @@ export const AddSetForm = ({
...
+      ordinal: 0,
+      minPoints: 0,
+    };
+
+    const levelsWithValues = [...levels, newLevel];
+    // check if array sorted by maxpoints has grades sorted as well
+    const levelsSortedByPoints = levelsWithValues.sort(
+      (l1, l2) => l1.maxPoints - l2.maxPoints,
+    );
+    const gradeError = levelsSortedByPoints.every((l1, index, array) => {
```
Comment (AnCichocka) [inline frontend/src/components/Edition/Sections/LevelsSection/AddSetForm/AddSetForm.tsx:126]: change to levels

**Review by AnCichocka — COMMENTED** [review]
🚀

**Review by Szyntos — APPROVED** [review]
yaay

**Pull request merged**